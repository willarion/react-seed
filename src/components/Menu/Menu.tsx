import React from 'react';
import { MenuItem } from '../MenuItem/MenuItem';

export const Menu: React.FC = ({}) => {
  //todo function toggleNavBar() {}

  return (
    <nav className="navbar navbar-default navbar-top">
      <div className="container">
        <div className="navbar-header">
          <button
            //todo onClick={toggleNavBar}
            type="button"
            className="navbar-toggle collapsed"
            data-toggle="collapse"
            data-target=".bs-example-navbar-collapse-1"
          >
            <span className="sr-only">Toggle navigation</span>
            <span className="icon-bar" />
            <span className="icon-bar" />
            <span className="icon-bar" />
          </button>
        </div>
        <div className="collapse navbar-collapse bs-example-navbar-collapse-1">
          <ul className="nav navbar-nav">
            <MenuItem
              to={'index.html'}
              itemName={'Menu Item'}
              isActive={false}
            />
            <MenuItem
              to={'index.html'}
              itemName={'Menu Item'}
              isActive={false}
            />
            <MenuItem
              to={'index.html'}
              itemName={'Menu Item'}
              isActive={false}
            />
            <MenuItem
              to={'index.html'}
              itemName={'Menu Item'}
              isActive={true}
            />
            <MenuItem
              to={'index.html'}
              itemName={'Menu Item'}
              isActive={false}
            />
          </ul>
        </div>
      </div>
    </nav>
  );
};
