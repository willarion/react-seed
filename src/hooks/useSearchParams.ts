import { useLocation } from 'react-router-dom';
import queryString from 'query-string';
import { MessagesSearchParams } from '../models/MessagesSearchParams';

export const useSearchParams = (): Record<
  string,
  string | Array<string> | null
> => {
  const location = useLocation();

  return queryString.parse(location.search);
};

export const useMessagesSearchString = (
  params?: Partial<MessagesSearchParams>,
): string => {
  const currentParams = useSearchParams();
  return queryString.stringify({ ...currentParams, ...params });
};
