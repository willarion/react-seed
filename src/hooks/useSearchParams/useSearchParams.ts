import { useLocation } from 'react-router-dom';
import queryString from 'query-string';
import { MessagesSearchParams } from '../../models/MessagesSearchParams';

export const useSearchParams = (): Record<
  string,
  string | Array<string> | null
> => {
  const location = useLocation();

  return queryString.parse(location.search);
};

export const useMessagesSearchString = (
  params?: Partial<MessagesSearchParams>,
): string => {
  const currentParams = useSearchParams();
  return queryString.stringify({ ...currentParams, ...params });
};

export const useResetCategories = (): string => {
  const currentParams = useSearchParams();
  const onlyLabes = {
    label: currentParams.label,
    search: currentParams.search,
  };
  return queryString.stringify(onlyLabes);
};

export const useResetLabels = (): string => {
  const currentParams = useSearchParams();
  const onlyCategories = {
    category: currentParams.category,
    search: currentParams.search,
  };
  return queryString.stringify(onlyCategories);
};

export const useResetSearch = (): string => {
  const currentParams = useSearchParams();
  const onlyLabes = {
    label: currentParams.label,
    categories: currentParams.categories,
  };
  return queryString.stringify(onlyLabes);
};
