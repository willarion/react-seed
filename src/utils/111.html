<!doctype html>
<html>
<head>
  <title>Postman | API Development Environment</title>
  <!-- <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,600,700" rel="stylesheet"> -->
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <!-- Marketo Variables -->
  <style type="text/css">

      /* latin-ext */
      @font-face {
          font-family: 'Open Sans';
          font-style: normal;
          font-weight: 400;
          src: local('Open Sans Regular'), local('OpenSans-Regular'), url(https://fonts.gstatic.com/s/opensans/v15/u-WUoqrET9fUeobQW7jkRYX0hVgzZQUfRDuZrPvH3D8.woff2) format('woff2');
          unicode-range: U+0100-024F, U+1E00-1EFF, U+20A0-20AB, U+20AD-20CF, U+2C60-2C7F, U+A720-A7FF;
      }
      /* latin */
      @font-face {
          font-family: 'Open Sans';
          font-style: normal;
          font-weight: 400;
          src: local('Open Sans Regular'), local('OpenSans-Regular'), url(https://fonts.gstatic.com/s/opensans/v15/cJZKeOuBrn4kERxqtaUH3ZBw1xU1rKptJj_0jans920.woff2) format('woff2');
          unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2212, U+2215;
      }
      /* latin-ext */
      @font-face {
          font-family: 'Open Sans';
          font-style: normal;
          font-weight: 600;
          src: local('Open Sans SemiBold'), local('OpenSans-SemiBold'), url(https://fonts.gstatic.com/s/opensans/v15/MTP_ySUJH_bn48VBG8sNSojoYw3YTyktCCer_ilOlhE.woff2) format('woff2');
          unicode-range: U+0100-024F, U+1E00-1EFF, U+20A0-20AB, U+20AD-20CF, U+2C60-2C7F, U+A720-A7FF;
      }
      /* latin */
      @font-face {
          font-family: 'Open Sans';
          font-style: normal;
          font-weight: 600;
          src: local('Open Sans SemiBold'), local('OpenSans-SemiBold'), url(https://fonts.gstatic.com/s/opensans/v15/MTP_ySUJH_bn48VBG8sNShampu5_7CjHW5spxoeN3Vs.woff2) format('woff2');
          unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2212, U+2215;
      }
      /* latin-ext */
      @font-face {
          font-family: 'Open Sans';
          font-style: normal;
          font-weight: 700;
          src: local('Open Sans Bold'), local('OpenSans-Bold'), url(https://fonts.gstatic.com/s/opensans/v15/k3k702ZOKiLJc3WVjuplzIjoYw3YTyktCCer_ilOlhE.woff2) format('woff2');
          unicode-range: U+0100-024F, U+1E00-1EFF, U+20A0-20AB, U+20AD-20CF, U+2C60-2C7F, U+A720-A7FF;
      }
      /* latin */
      @font-face {
          font-family: 'Open Sans';
          font-style: normal;
          font-weight: 700;
          src: local('Open Sans Bold'), local('OpenSans-Bold'), url(https://fonts.gstatic.com/s/opensans/v15/k3k702ZOKiLJc3WVjuplzBampu5_7CjHW5spxoeN3Vs.woff2) format('woff2');
          unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2212, U+2215;
      }

      #mkto-header-image {
          width: 120px !important;
      }
      .show {
          display: block !important;
      }

      .hide {
          display: hidden !important;
      }
      .bodyContent a {
          color: #ff6c37 !important;
      }
      #titleTxt {
          text-align: center !important;
      }
      .content-img {
          width: 100% !important;
          display: block !important;
          margin: 0 auto !important;
      }
      .content-img img {
          display: inline-block !important;
          width: 90%;
          height: auto !important;
          text-align: center !important;
          margin: 0 auto !important;
      }
      #footerText {
          color: #666666 !important;
          font-size: 10px !important;
      }

      #footerText a {
          color: #666666 !important;
      }
      .h1Title {
          font-size: 32px;
          line-height: 1.1em;
          margin: 0px 50px !important;
      }

      .bodyContent {
          margin: 0px 50px;
          font-size: 14px;
          line-height: 18px;
          font-weight: 400;
          color: #666666;
      }

      .button {
          background-color: #ff6c37;
          border: 1px solid #ff6c37;
          color: #FFFFFF !important;
      }
      .button:hover {
          background-color: #EF5B25;
      }

      @media only screen and (max-width: 600px){
          .h1Title{
              font-size:28px !important;
              margin: 0px 10px !important;
          }
          .bodyContent {
              font-size: 16px !important;
              margin: 0px 10px !important;
          }
      }

  </style>
</head>
<!-- [if !((gte mso 9)|(IE))]><!-->
<body bgcolor="#fafafa" style="font-family:'Open Sans', -apple-system, Helvetica, Arial, sans-serif !important;"><style type="text/css">div#emailPreHeader{ display: none !important; }</style><div id="emailPreHeader" style="mso-hide:all; visibility:hidden; opacity:0; color:transparent; mso-line-height-rule:exactly; line-height:0; font-size:0px; overflow:hidden; border-width:0; display:none !important;">Youâ€™re shaping the API industry, and we want to hear from you. </div>
<!-- Pre-header - set this hidden -->
<div style="display: none !important; width:0; height:0; overflow:hidden; float:none; visibility:hidden; line-height:0; font-size:0; max-height:0; mso-hide: all"></div>
<!-- <![endif] -->
<!-- Email Content -->
<table id="body" class="body" summary="body" align="center" border="0" cellspacing="0" cellpadding="0" style="width: 100%; height: 100%; background-color: #fafafa; margin: 20px 0;">
  <tbody>
  <tr>
    <td align="center" valign="top">
      <table class="outer-table mktoContainer" id="template-wrapper" summary="outer-table" align="center" valign="top" border="0" cellpadding="0" cellspacing="0" bgcolor="#FFFFFF" style="background-color: #FFFFFF !important; width: 100%; max-width: 650px; margin: 0 auto;">
        <tr id="email-header" class="mktoModule">
          <td>
            <table align="center" valign="top" border="0" cellpadding="0" cellspacing="0" style="width: 100%;background-color: #FFFFFF; margin: 0 auto; text-align:center;">
              <tbody>
              <tr>
                <td align="center" valign="top" style="background-color: #FFFFFF;">
                  <div class="mktoImg" id="mkto-header-image" style="margin: 25px 0 20px;">
                    <img src="https://pages.getpostman.com/rs/067-UMD-991/images/pm-logo-horiz201912.png" width="130" style="border:0; width:130px;">
                  </div> </td>
              </tr>
              </tbody>
            </table> </td>
        </tr>
        <tr id="title" class="mktoModule">
          <td style="display: block !important;">
            <div id="titleTxt" class="h1Title" style="color: #282828 !important; font-family:'Open Sans', -apple-system, Helvetica, Arial, sans-serif !important; font-weight: 600; text-align: center !important;">
              Two Days Left to Take the Survey for Your Chance to Win
            </div> </td>
        </tr>
        <tr height="10" class="mktoModule" id="spacer-10fb4e2d58-bcea-49de-9c8e-6a935b13516d7a27a3fe-f7fb-45a9-931d-1361037e2ed43a100bff-65c1-4812-ac64-5f7e294d3dd9" style="background-color: transparent !important;" border="0">
          <td height="20" style="border-spacing: 0 !important; font-size: 0px !important; padding: 0px; margin: 0px; border: none;" border="0">&nbsp;</td>
        </tr>
        <tr id="email-image76badb18-3e92-4b27-a563-3217cfe865ca" class="mktoModule show" align="center">
          <td style="display: block !important;">
            <div class="mktoImg content-img" id="mkto-email-image76badb18-3e92-4b27-a563-3217cfe865ca" style="background-color: #FFFFFF; text-align: center !important; margin: 0 auto;" mktolockimgsize="true">
              <img src="https://lp.postman.com/rs/067-UMD-991/images/bann-email-reengagment-Publishing-R3.png" style="">
            </div> </td>
        </tr>
        <tr height="10" class="mktoModule" id="spacer-10fb4e2d58-bcea-49de-9c8e-6a935b13516d" style="background-color: transparent !important;" border="0">
          <td height="20" style="border-spacing: 0 !important; font-size: 0px !important; padding: 0px; margin: 0px; border: none;" border="0">&nbsp;</td>
        </tr>
        <tr id="body" class="mktoModule">
          <td>
            <div class="mktoText bodyContent" id="bodyTxt" style="text-align: center;">
              <p style="line-height: 1.2; margin-top: 0pt; margin-bottom: 0pt; text-align: left;"><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre-wrap;">You're charting the future of APIs, and we want to know what you think.</span></p>
              <p style="line-height: 1.2; margin-top: 0pt; margin-bottom: 0pt; text-align: left;"><br></p>
              <p style="line-height: 1.2; margin-top: 0pt; margin-bottom: 0pt; text-align: left;"><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre-wrap;">It only takes about 15 minutes to complete the </span><a href=
                                                                                                                                                                                                                                                                                                                                                                               "https://go.postman.com/MDY3LVVNRC05OTEAAAF-Z0gQ2Dq-Q6srxcKuaOBrDyRRarlNBVD3FJ_ghbQGSa6UhRRL-zRKyPaO97vk6thONsg0Ua6dmlNLyq4=" style="text-decoration: none;"
              ><span style="font-size: 11pt; font-family: Arial; color: #1155cc; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: underline; -webkit-text-decoration-skip: none; text-decoration-skip-ink: none; vertical-align: baseline; white-space: pre-wrap;">2021 State of the API survey</span></a><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre-wrap;">. Plus, if youâ€™re interested, weâ€™ll share the final report with you.</span></p>
              <p style="line-height: 1.2; margin-top: 0pt; margin-bottom: 0pt; text-align: left;"><br></p>
              <p style="line-height: 1.2; margin-top: 0pt; margin-bottom: 0pt; text-align: left;"><b><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre-wrap;">Only two days left to win prizes.</span></b></p>
              <p style="line-height: 1.2; margin-top: 0pt; margin-bottom: 0pt; text-align: left;"><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre-wrap;">At the end of the survey, you'll have a chance to enter our prize drawing which includes:</span></p>
              <ul style="margin-top: 0; margin-bottom: 0; padding-inline-start: 48px;">
                <li style="list-style-type: disc; font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre; text-align: left;"><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration:
none; vertical-align: baseline; white-space: pre-wrap;">Sony PlayStationÂ®5 (1 winner)</span></li>
                <li style="list-style-type: disc; font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre; text-align: left;"><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration:
none; vertical-align: baseline; white-space: pre-wrap;">$100 Amazon gift card (5 winners)</span></li>
                <li style="list-style-type: disc; font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration: none; vertical-align: baseline; white-space: pre; text-align: left;"><span style="font-size: 11pt; font-family: Arial; color: #000000; background-color: transparent; font-weight: 400; font-variant: normal; text-decoration:
none; vertical-align: baseline; white-space: pre-wrap;">$50 gift certificate to the Postman swag store (10 winners)</span></li>
              </ul>
            </div> </td>
        </tr>
        <tr height="10" class="mktoModule" id="spacer-10fb4e2d58-bcea-49de-9c8e-6a935b13516d7a27a3fe-f7fb-45a9-931d-1361037e2ed4" style="background-color: transparent !important;" border="0">
          <td height="20" style="border-spacing: 0 !important; font-size: 0px !important; padding: 0px; margin: 0px; border: none;" border="0">&nbsp;</td>
        </tr>
        <tr id="button841ab662-667f-41e8-ae32-e8541e8ce1d1" class="mktoModule">
          <td align="center">
            <table width="100%" border="0" cellspacing="0" cellpadding="0">
              <tbody>
              <tr>
                <td align="center">
                  <table border="0" cellspacing="0" cellpadding="0">
                    <tbody>
                    <tr>
                      <td align="center"> <a href=
                                               "https://go.postman.com/MDY3LVVNRC05OTEAAAF-Z0gQ2EYqIQyS_dyGCoezaCh1iJexMidWpLNVoWMTQu4HBQDIW0j57Fnia5cmQ66TRyRZEzPuG-m8KC8=" id="cta-btn" mktoname="CTA Button" summary="main-cta" target="_blank" class="button" style="font-size: 16px; line-height: 22px !important; font-weight: 600!important; color: #ffffff !important; text-decoration: none; text-decoration: none;border-radius: 3px; padding: 12px 40px; display: inline-block;font-family:'Open Sans', -apple-system, Helvetica, Arial, sans-serif !important;"
                      >Take the Survey</a> </td>
                    </tr>
                    </tbody>
                  </table> </td>
              </tr>
              </tbody>
            </table> </td>
        </tr>
        <tr class="mktoModule" id="footer">
          <td>
            <table class="inner-table" summary="inner-table" align="center" valign="top" border="0" cellpadding="0" cellspacing="0" style="width: 100%;margin: 0 auto; text-align:center;" bgcolor="#FFFFFF">
              <tbody>
              <tr height="136">
                <td>
                  <div class="mktoText" id="footerText">
                    <p style="font-family: 'Open Sans', -apple-system, Helvetica, Arial, sans-serif !important; font-size: 10px; line-height: 14px; color: #666666;">Â© 2021 Postman Inc., All Rights Reserved.<br>55 2nd Street, Suite 300 San Francisco, CA 94105, USA<br></p>
                  </div> </td>
              </tr>
              </tbody>
            </table> </td>
        </tr>
        <tr class="mktoModule" id="unsubfooter">
          <td>
            <table class="inner-table" summary="inner-table" align="center" valign="top" border="0" cellpadding="0" cellspacing="0" style="width: 100%;margin: 0 auto; text-align:center;" bgcolor="#fafafa">
              <tbody>
              <tr>
                <td> <p style="font-family: 'Open Sans', -apple-system, Helvetica, Arial, sans-serif !important; font-size: 10px; margin: 30px 10px; line-height: 14px; color: #666666; !important;">Click to <strong><a href="https://lp.postman.com/UnsubscribePage.html?mkt_unsubscribe=1&mkt_tok=MDY3LVVNRC05OTEAAAF-Z0gQ2ClkxdbQdJ9wB0rcuUoWVKICY-mpIsivaaKc0v_qMnM-KZHPyW6WjgbzUH5wmmMo58CJ81XNuy2XU0VsTKQrwkB1pcVOVSFD8bcVlnryBIL6dAMY" style="color: #666666 !important;">unsubscribe</a></strong> from these emails.</p> </td>
              </tr>
              </tbody>
            </table> </td>
        </tr>
      </table> </td>
  </tr>
  </tbody>
</table>

<img src="https://go.postman.com/trk?t=1&mid=MDY3LVVNRC05OTE6MDo1Nzg3OjU3MDczOjM1OTI1MTo4MDU4Ojk6Mzc5OTY6NTc5OTkwMjYtMC0yNDc6YW5pc2ltb3ZhLm1hcmluYS5uQGdtYWlsLmNvbQ%3D%3D" width="1" height="1" style="display:none !important;" alt="" />
</body>
</html>