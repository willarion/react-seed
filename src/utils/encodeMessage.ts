import { MessageFormInput } from '../models/MessageFormInput';
import base64url from 'base64url';

export const encodeMessage = ({
  to,
  copy,
  secret,
  subject,
  message,
}: MessageFormInput): string => {
  const plainMessage = `To: <${to}>
Cc: ${copy}
Bcc: ${secret}
Subject: ${subject}

${message}`;

  return base64url(plainMessage);
};
