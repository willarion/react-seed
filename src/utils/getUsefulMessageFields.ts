import { GoogleMessage } from '../models/GoogleMessage';
import { MessageToRender } from '../models/MessageToRender';

interface Header {
  name: string;
  value: string;
}

export function getUsefulMessageFields(
  item: GoogleMessage,
  snippet: boolean,
): MessageToRender {
  function filter(array: Header[], key: string) {
    const result = array.find((header: Header) => header.name === key);
    return result?.value || '';
  }

  const title = filter(item.payload.headers, 'Subject');
  const date = filter(item.payload.headers, 'Date');
  const from = filter(item.payload.headers, 'From');
  // checking is there anything else except snippet
  const fullText = item.payload.parts[0].body.data
    ? item.payload.parts[0].body.data
    : item.snippet;
  const text = snippet ? item.snippet : fullText;

  return {
    text,
    title,
    date,
    from,
    id: item.id,
  };
}
