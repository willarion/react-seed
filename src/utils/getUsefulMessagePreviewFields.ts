import { GoogleMessage } from '../models/GoogleMessage';
import { UserPreviewMessage } from '../models/UserPreviewMessage';
import { find, get } from 'lodash';
import { decode } from 'js-base64';
import DOMPurify from 'dompurify';
import he from 'he';
import { UserFulltextMessage } from '../models/UserFulltextMessage';

export const getUsefulMessagePreviewFields = (
  message: GoogleMessage,
): UserPreviewMessage => {
  const title = find(message.payload.headers, { name: 'Subject' })?.value;
  const date = find(message.payload.headers, { name: 'Date' })?.value;
  const from = find(message.payload.headers, { name: 'From' })?.value;
  const text = he.decode(message.snippet);

  return {
    text: text,
    title,
    date,
    from,
    id: message.id,
  };
};

export const getUsefullFulltextMessageFields = (
  message: GoogleMessage,
): UserFulltextMessage | null => {
  if (!get(message, message.payload.parts[1].body.data)) {
    return null;
  }

  const { title, date, from, id } = getUsefulMessagePreviewFields(message);
  const encodedHTML = get(message, message.payload.parts[1].body.data);
  const messageDangerHTML = decode(encodedHTML);
  const messageSafeHTML = DOMPurify.sanitize(messageDangerHTML, {
    FORCE_BODY: true,
  });

  return {
    title,
    date,
    from,
    id,
    html: messageSafeHTML,
  };
};
