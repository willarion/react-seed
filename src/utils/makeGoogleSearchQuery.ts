import { GoogleMessageSearch } from '../models/GoogleMessageSearch';
import { ParsedQuery } from 'query-string';

export const makeGoogleSearchParams = (
  search: ParsedQuery<string>,
): Partial<GoogleMessageSearch> => {
  let category: GoogleMessageSearch['category'];
  let is: GoogleMessageSearch['label'];
  let q: GoogleMessageSearch['searchInput'];

  if (search.category) {
    switch (search.category) {
      case 'promotions':
        category = 'category:promotions';
        break;
      case 'social':
        category = 'category:social';
        break;
    }
  }
  if (search.label) {
    switch (search.label) {
      case 'inbox':
        is = 'in:inbox';
        break;
      case 'drafts':
        is = 'in:draft';
        break;
      case 'starred':
        is = 'is:starred';
        break;
    }
  }
  if (search.search) {
    q = search.search;
  }

  return { category, label: is, searchInput: q };
};

export const makeGoogleSearchQuery = (search: ParsedQuery<string>): string => {
  const { category, label, searchInput } = makeGoogleSearchParams(search);

  return [category, label, searchInput].join(' ');
};
