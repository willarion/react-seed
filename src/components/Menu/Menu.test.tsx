import { shallow } from 'enzyme';
import { Menu } from './Menu';
import { useMessagesSearchString } from '../../hooks/useSearchParams/useSearchParams';
import { mocked } from 'ts-jest/utils';

jest.mock('../../hooks/useSearchParams/useSearchParams');

const mockedHook = mocked(useMessagesSearchString).mockImplementation(
  () => 'some string',
);

describe('<Menu />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should have render', () => {
    const wrapper = shallow(<Menu />);
    expect(wrapper).toMatchSnapshot();
    expect(mockedHook).toBeCalledTimes(2);
  });
});
