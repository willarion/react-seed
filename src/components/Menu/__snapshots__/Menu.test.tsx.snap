// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should have render 1`] = `
<nav
  className="navbar navbar-default navbar-top"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-toggle collapsed"
        data-target=".bs-example-navbar-collapse-1"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      className="collapse navbar-collapse bs-example-navbar-collapse-1"
    >
      <ul
        className="nav navbar-nav"
      >
        <MenuItem
          search="some string"
          to="/home"
        >
          Promotions
        </MenuItem>
        <MenuItem
          search="some string"
          to="/home"
        >
          Social
        </MenuItem>
      </ul>
    </div>
  </div>
</nav>
`;
